warehouse/internal/repo/schema.go

```diff
@@ -146,6 +257,94 @@ func (sh *WHSchema) GetForNamespace(ctx context.Context, destID, namespace strin
 	return *entries[0], err
 }
 
+// populateTableLevelSchemasWithTx inserts table-level schemas for each table in the parent schema
+// that don't already exist as separate table-level schemas, using the provided transaction.
+func (sh *WHSchema) populateTableLevelSchemasWithTx(ctx context.Context, tx *sqlmiddleware.Tx, destID, namespace string) error {
+	query := `
+		INSERT INTO wh_schemas (
+			source_id,
+			namespace,
+			destination_id,
+			destination_type,
+			schema,
+			table_name,
+			created_at,
+			updated_at,
+			expires_at
+		)
+		SELECT
+			s.source_id,
+			s.namespace,
+			s.destination_id,
+			s.destination_type,
+			j.value AS schema,
+			j.key AS table_name,
+			s.created_at,
+			s.updated_at,
+			s.expires_at
+		FROM wh_schemas s
+		CROSS JOIN LATERAL jsonb_each(s.schema) AS j
+		WHERE
+			s.destination_id = $1 AND
+			s.namespace = $2 AND
+			s.table_name = '' AND
+			NOT EXISTS (
+				SELECT 1 FROM wh_schemas s2
+				WHERE s2.source_id = s.source_id AND
+					s2.namespace = s.namespace AND
+					s2.destination_id = s.destination_id AND
+					s2.table_name = j.key
+			)
+	`
+	_, err := tx.ExecContext(ctx, query, destID, namespace)
+	if err != nil {
+		return fmt.Errorf("populating table-level schemas: %w", err)
+	}
+	return nil
+}
+
+// getTableLevelSchemasForNamespaceWithTx fetches and merges all table-level schemas for destID and namespace,
+// using the provided transaction.
+func (sh *WHSchema) getTableLevelSchemasForNamespaceWithTx(ctx context.Context, tx *sqlmiddleware.Tx, destID, namespace string) (model.Schema, error) {
+	tableLevelQuery := `SELECT table_name, schema FROM ` + whSchemaTableName + `
+	WHERE
+		destination_id = $1 AND
+		namespace = $2 AND
+		table_name != ''
+	ORDER BY table_name;
```

Comment:
- Why do we need a `ORDER BY` ?
- There can be multiple table level schemas if there are multiple sources connected to the destination. Are we handling that case (LIMIT 1) ?

Reply:
Ordering is not required since all of the sources will have the same table schema. Removing it.